service/listeners.go

```diff
@@ -287,10 +278,16 @@ func (m *multiPacketListener) Acquire() (net.PacketConn, error) {
 		m.readCh = make(chan readRequest)
 		m.doneCh = make(chan struct{})
 		go func() {
+			buffer := make([]byte, serverUDPBufferSize)
 			for {
+				n, addr, err := m.pc.ReadFrom(buffer)
+				if errors.Is(err, net.ErrClosed) {
+					return
```

Comment:
Let's close the readCh when this happens. Perhaps best done as a defer in the beginning of the go routine.

Then in `virtualPacketConn.ReadFrom` let's check if the channel read is successful (not closed).

This ensures the ReadFroms always return, even if we don't call close, since we created a case here where we may stop serving before the clients closed.

Reply:
I'm not sure we can check the channel read is successful in `virtualPacketConn.ReadFrom()`.

I think actually we can move this down so that we continue to send the closed error into the read channel for `virtualPacketConn`s so they return, and then we return and close the channel here. Wdyt?

Reply:
As discussed offline, removing the `close(m.readCh)`.